walk-in-interview/src/main/java/com/google/job/servlets/JobServlet.java

```diff
@@ -123,12 +152,12 @@ private Job parseRawJobPost(HttpServletRequest request) throws IOException, Ille
         }
     }
 
-    /** Stores the job post into the database. */
-    private void storeJobPost(Job job) throws ServletException, ExecutionException, TimeoutException {
+    /** Stores the job post into the database and updates business account accordingly. */
+    private void storeJobPost(String uid, Job job) throws ServletException, ExecutionException, TimeoutException {
         try {
             // Blocks the operation.
             // Use timeout in case it blocks forever.
-            this.jobsDatabase.addJob(job).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            this.jobsDatabase.addJob(uid, job).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
```

Comment:
Maybe call businessDatabase.updateJobsMade(uid, jobId); here so job database doesn't need to depend on business database?

Reply:
`seems cannot, since still not find a way to get jobId once it is newly created; for newly created Job, it returns Future<WriteResult>, cannot get a jobId`